extensions/wikia/Wall/builders/WallMessageBuilder.class.php

```diff
@@ -0,0 +1,330 @@
+	private $messageText;
+	/** @var User $messageAuthor */
+	private $messageAuthor;
+	/** @var Title $parentPageTitle */
+	private $parentPageTitle;
+	/** @var string $messageTitle */
+	private $messageTitle;
+
+	// Optional fields
+	/** @var WallMessage|null $parentMessage */
+	private $parentMessage = null;
+	/** @var bool $notify; */
+	private $notify = true;
+	/** @var bool $notifyEveryone */
+	private $notifyEveryone = false;
+	/** @var array $relatedTopics */
+	private $relatedTopics = [];
+
+	// Generated fields
+	/** @var array $metaData */
+	private $metaData;
+	/** @var WallMessage $newMessage */
+	private $newMessage;
+
+	/**
+	 * Check if Forum Board or Message Wall we are posting on exists, and try to create it if does not yet exist.
+	 * @return WallMessageBuilder
+	 */
+	public function createParentPageIfNotExists(): WallMessageBuilder {
+		if ( !$this->parentPageTitle->exists() ) {
+			$robot = User::newFromName( Wikia::BOT_USER );
+			$page = WikiPage::factory( $this->parentPageTitle );
+
+			$status = $page->doEdit( '', '', EDIT_NEW | EDIT_MINOR | EDIT_SUPPRESS_RC | EDIT_FORCE_BOT, false, $robot );
+			if ( !$status->isOK() ) {
+				$this->throwException( 'Failed to create parent page for message' );
+			}
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Given user-supplied parameters, initialize message metadata which will be saved as part of comment content.
+	 * @return WallMessageBuilder
+	 */
+	public function createMetaData(): WallMessageBuilder {
+		$this->metaData = [ 'title' => $this->messageTitle ];
+		if ( $this->notifyEveryone ) {
+			$this->metaData[ 'notify_everyone' ] = time();
+		}
+		if ( !empty( $this->relatedTopics ) ) {
+			$this->metaData[ 'related_topics' ] = implode( '|', $this->relatedTopics );
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Call ArticleComments and actually save the new message.
+	 * @return WallMessageBuilder
+	 */
+	public function postNewMessage(): WallMessageBuilder {
+		/**
+		 * Hook into MW transaction to ensure data integrity is maintained in comments_index table
+		 * If handler cannot create comments_index entry, transaction will be aborted and comment will not be posted!
+		 * @see WallMessageBuilder::insertNewCommentsIndexEntry()
+		 */
+		Hooks::register( 'ArticleDoEdit', [ $this, 'insertNewCommentsIndexEntry' ] );
+
+		if ( !$this->parentMessage ) {
+			$result = ArticleComment::doPost( $this->messageText, $this->messageAuthor, $this->parentPageTitle, false, $this->metaData );
+		} elseif ( !$this->parentMessage->canReply() ) {
+			$this->throwException( 'Attempted to post reply on deleted or removed thread' );
+		} else {
+			$result = ArticleComment::doPost( $this->messageText, $this->messageAuthor, $this->parentMessage->getTitle(), $this->parentMessage->getId() );
+		}
+
+		if ( !$result || !$result[0]->isOK() ) {
+			$this->throwException( 'Failed to create article comment' );
+		}
+
+		/** @var Article $article */
+		list( $status, $article ) = $result;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$status" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)